snapcraft_parser/main.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/python3
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+snapcraft-parser
+
+Usage:
+  snapcraft-parser [options]
+
+Options:
+  -h --help                             show this help message and exit
+  -p --print                            print the master parts list
+  -v --version                          show program version and exit
+  -d --debug                            print debug information while executing
+                                        (including backtraces)
+
+
+For more help, visit the documentation:
+http://developer.ubuntu.com/snappy/snapcraft
+"""
+
+import gzip
+import logging
+import os
+import pkg_resources
+import sys
+import textwrap
+import yaml
+
+from docopt import docopt
+
+import snapcraft_parser
+from snapcraft.internal import log, sources
+
+
+logger = logging.getLogger(__name__)
+
+
+PART_NAMESPACE_SEP = '/'
+# TODO: make this a temporary directory that get's removed when finished
+BASE_DIR = "/tmp"
+PARTS_FILE = os.path.join(BASE_DIR, "snap-parts.yaml.gz")
```

Comment:
I've added the "--output" flag which allows the user to provide the location of the parts list and the default is now the current directory.